### PR TITLE
Fix favorite course

### PR DESCRIPTION
**Pull request by ChiWeiHsiao**
*  Modify course add_to_cart
    * Modify the method "add_to_cart" in course_controller from cookie to many_to_many
    * Add new method "destroy_cart" to destroy favorite courses

*  Add RSpec for user's favorite courses 
    *  spec/models/user_course_detailship_spec.rb
    *  spec/controllers/courses_controller_spec.rb

*  Add new tip of how to delete courses in cart

### CODE REVIEW

File: app/models/user_course_detailship.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class UserCourseDetailship < ActiveRecord::Base
```
Comment (t6847kimo) [inline app/models/user_course_detailship.rb]: Please re-name it. Such as user_favorite_courses.

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -1,174 +1,172 @@
...
-				cd.to_search_result
-			}
-		}
-		render :layout=>false
-	end
-		
-  
+
+  def destroy_cart
+    current_user.user_course_detailships.destroy_all
```
Comment (t6847kimo) [inline app/controllers/courses_controller.rb]: Are you sure user want to destroy all?

Comment (ChiWeiHsiao) [inline app/controllers/courses_controller.rb]: I removed this function in the next commit (Add RSpec for user's favorite courses).
Sorry for the unclear commit message. QQ

File: app/models/user_course_detailship.rb (hunk 1 above)
Comment (ChiWeiHsiao) [inline app/models/user_course_detailship.rb]: I just renamed it, please help me to review again. Thanks~

**Comment by t6847kimo** [comment]
OK

**Pull request merged**